### ISSUE TITLE
DataTable's style_* should allow class names instead of CSS styles

### ISSUE THREAD

**Issue by quazgar** [plotly/dash-table#870]
It should be possible to specify class names instead of verbatim CSS style attributes in the `style_*` parameters of DataTable.  This way, styling could be done at a later time and independent from the table creation code.

## API change proposal ##

Class names could be given as lists of strings, in the case of `style_*_conditional`, there could be a second special key `classNames` which takes a list of strings.

```py
DataTable(...,
    style_header=['myHeaderClass', 'myTableClass'],
    style_data_conditional=[
        {
            'if': {'row_index': 'odd'},
            'classNames': ['myOddClass', 'myTableClass'],
            },
    ...
)
```

Thanks for your interest in Plotly's Dash DataTable component!!

Note that GitHub issues in this repo are reserved for bug reports and feature
requests. Implementation questions should be discussed in our
[Dash Community Forum](https://community.plotly.com/c/dash).

Before opening a new issue, please search through existing issues (including
closed issues) and the [Dash Community Forum](https://community.plotly.com/c/dash).
If your problem or idea has not been addressed yet, feel free to
[open an issue](https://github.com/plotly/plotly.py/issues/new).

When reporting a bug, please include a reproducible example! We recommend using
the [latest version](https://github.com/plotly/dash-table/blob/master/CHANGELOG.md)
as this project is frequently updated. Issues can be browser-specific so
it's usually helpful to mention the browser and version that you are using.

Thanks for taking the time to help up improve this component!

**Comment by alexcjohnson** [comment]
That's a great idea @quazgar - but is it important that this be a list rather than a space-separated string? That's what we use throughout DCC, it's a more direct line to the underlying HTML, and it's simpler when you just want a single class.

**Comment by quazgar** [comment]
No, that's no problem, whatever is easier to implement. Actually, maybe the two ways to specify styling could be unified, by always using the special key:

```py
DataTable(...,
    style_header={
        'textAlign': 'left',
        'classNames': 'myHeaderClass myTableClass'
    },
    style_data_conditional=[
        {
            'if': {'row_index': 'odd'},
            'classNames': 'myOddClass myTableClass',
        },
    ...
)
```

**Comment by alexcjohnson** [comment]
Yes exactly, apply this to the conditional and static style props alike. We'd still support explicit styling, just allow the special key `className` that gets passed along to the class attribute instead of the style attribute. Note it should be `className` not `classNames` - to match how [react](https://reactjs.org/docs/faq-styling.html) does it which is the convention DCC and dash-html-components followed.

**Comment by ned2** [comment]
I would be so keen to see this feature added. Just found myself scouring the DataTable reference hoping I would find it. Found this issue via an existing [StackOverflow question](https://stackoverflow.com/questions/70378819/dashplotly-conditional-formatting-append-a-classname) where someone else was looking for this feature also.  

Are there any dependencies for adding this feature, or is it just a matter of prioritisation @alexcjohnson?

edit: oops, just realised I'm in the wrong repo

**Comment by a4shubert** [comment]
I would be very keen to have this feature. It's just not clear how to structure the styling of dash application: you can use properly created external css file (which in turn could be created by sass etc) for all the html components, but for the dash-table you have to copy/paste all the different values from already existing css into the string.

**Comment by BShuYi** [comment]
I would be very keen to have this feature. So, same design can apply to multiple table without repeat all the style and can reduce the script of it. Hope have this feature soon.

**Comment by frnhr** [comment]
One thing that CSS stylesheets can do that inline style cannot is play with pseudo-elements and pseudo-classes (`:hover`, `::before`, etc.) And another thing that is not supported via Dash's `style*` attributes is styling child elements inside cells.

Would be great to have this conditional className implemented!
